File: java/com/countrygamer/countrygamercore/common/inventory/FakeInventory.java
package com.countrygamer.countrygamercore.common.inventory;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.inventory.IInventory;
import net.minecraft.item.ItemStack;

public class FakeInventory implements IInventory {
	
	private ItemStack holderStack;
	private ItemStack[] inv;
	private final int invSize;
	private final int stackLimit;
	
	public FakeInventory(ItemStack itemStack, int inventorySize, int maxStackSizePerSlot) {
		this.holderStack = itemStack;
		this.invSize = inventorySize;
		this.inv = new ItemStack[this.invSize];
		this.stackLimit = maxStackSizePerSlot;
		
	}
	
	@Override
	public int getSizeInventory() {
		return this.invSize;
	}
	
	@Override
	public ItemStack getStackInSlot(int slot) {
		return this.inv[slot];
	}
	
	@Override
	public ItemStack decrStackSize(int slot, int amount) {
		if (this.inv[slot] != null) {
			ItemStack itemstack;
			
			if (this.inv[slot].stackSize <= amount) {
				itemstack = this.inv[slot];
				this.inv[slot] = null;
				this.markDirty();
				return itemstack;
			}
			else {
				itemstack = this.inv[slot].splitStack(amount);
				
				if (this.inv[slot].stackSize == 0) {
					this.inv[slot] = null;
				}
				
				this.markDirty();
				return itemstack;
			}
		}
		else {
			return null;
		}
	}
	
	@Override
	public ItemStack getStackInSlotOnClosing(int slot) {
		return this.getStackInSlot(slot);
	}
	
	@Override
	public void setInventorySlotContents(int slot, ItemStack stack) {
		this.inv[slot] = stack;
		this.markDirty();
	}
	
	@Override
	public String getInventoryName() {
		return this.holderStack != null ? this.holderStack.getDisplayName() : "";
	}
	
	@Override
	public boolean hasCustomInventoryName() {
		return false;
	}
	
	@Override
	public int getInventoryStackLimit() {
		return this.stackLimit;
	}
	
	@Override
	public void markDirty() {
		
	}
	
	@Override
	public boolean isUseableByPlayer(EntityPlayer var1) {
		return true;
	}
	
	@Override
	public void openInventory() {
		
	}
	
	@Override
	public void closeInventory() {
		
	}
	
	@Override
	public boolean isItemValidForSlot(int slot, ItemStack stack) {
		return true;
	}
	
}

File: src/main/scala/com/countrygamer/cgo/wrapper/common/inventory/FakeInventory.scala
package com.countrygamer.cgo.wrapper.common.inventory

import net.minecraft.entity.player.EntityPlayer
import net.minecraft.inventory.IInventory
import net.minecraft.item.ItemStack

/**
 * A default fake inventory for easy access
 *
 * @param holderStack
 * The itemStack used to instantiate this inventory
 * @param inventorySize
 * The size of this inventory (number of slots)
 * @param stackLimit
 * The maximum stack size per slot
 *
 * @author CountryGamer
 */
class FakeInventory(var holderStack: ItemStack, var inventorySize: Int, var stackLimit: Int)
		extends IInventory {

	// Default Constructor
	private val inv: Array[ItemStack] = new Array[ItemStack](this.inventorySize)
	// End Constructor

	def getSizeInventory: Int = {
		return this.inventorySize
	}

	def getStackInSlot(slot: Int): ItemStack = {
		return this.inv(slot)
	}

	def decrStackSize(slot: Int, amount: Int): ItemStack = {
		if (this.inv(slot) != null) {
			var itemstack: ItemStack = null
			if (this.inv(slot).stackSize <= amount) {
				itemstack = this.inv(slot)
				this.inv(slot) = null
				this.markDirty
				return itemstack
			}
			else {
				itemstack = this.inv(slot).splitStack(amount)
				if (this.inv(slot).stackSize == 0) {
					this.inv(slot) = null
				}
				this.markDirty
				return itemstack
			}
		}
		else {
			return null
		}
	}

	def getStackInSlotOnClosing(slot: Int): ItemStack = {
		return this.getStackInSlot(slot)
	}

	def setInventorySlotContents(slot: Int, stack: ItemStack) {
		this.inv(slot) = stack
		this.markDirty
	}

	def getInventoryName: String = {
		return if (this.holderStack != null) this.holderStack.getDisplayName else ""
	}

	def hasCustomInventoryName: Boolean = {
		return false
	}

	def getInventoryStackLimit: Int = {
		return this.stackLimit
	}

	def markDirty {
	}

	def isUseableByPlayer(var1: EntityPlayer): Boolean = {
		return true
	}

	def openInventory {
	}

	def closeInventory {
	}

	def isItemValidForSlot(slot: Int, stack: ItemStack): Boolean = {
		return true
	}

}

